elegant/segment_images.py

```diff
@@ -122,6 +122,8 @@ def annotate_poses_from_masks(positions, mask_root, annotations, overwrite_exist
             the worm hatched at the first timepoint taken.
     """
     for position_name, timepoint_name, image_path in load_data.flatten_positions(positions):
+        print(f'Annotating poses for position {position_name}')
```

Comment:
What's the use-case here? Just watching to see how fast it goes? Or providing debug info when it chokes on a particular image? If primarily the latter, it would be better (especially in the context of non-interactive uses) to just make sure the position name gets kicked up as part of the exception message. I.e. catch the exception and raise a new one with the relevant information including position name. If the exceptions get properly logged (as above), then this will make debugging easier.

Reply:
Mostly the former. A reasonable example is a user running segmentation in bulk on a large number of images (which couldn't have been done during the experiment because of the sheer number of images acquired) where calculating+storing annotations takes a sufficiently long time. In this case, progress updates make it feasible to track whether the annotation is in some bad state or just taking a long time. Otherwise, the user is left to hope that this already long computation is still going and that the user isn't sinking more hours into zero gain.

Reply:
OK, seems reasonable.